### PR TITLE
Select2Choice and Select2MultiChoice have to retain value after page get refreshed...

### PR DESCRIPTION
**Pull request by shuraa**
I believe, that issue #390 should be fixed. I've added several tests that covers several scenarios: 
* select2 retain value after page refresh
* select2 retain value if form wasn't validated

### CODE REVIEW

**Comment by solomax** [comment]
Why the formatting is changed again?
Maybe this can be moved to the separated PR? (I actually thought it was already applied)

File: jdk-1.6-parent/select2-parent/select2/src/main/java/org/wicketstuff/select2/AbstractSelect2Choice.java [hunk 1]
```diff
@@ -257,12 +286,35 @@ public void renderHead(IHeaderResponse response)
...
+	 */
+	protected final M getCurrentValue()
+	{
+		if (hasRawInput())
+		{
+			// since raw input is retained, we need explicitly convert it to target value
+			if (convertValuePerformed)
+			{
+				return getConvertedInput();
+			} else
```
Comment (solomax) [inline jdk-1.6-parent/select2-parent/select2/src/main/java/org/wicketstuff/select2/AbstractSelect2Choice.java]: formatting, 'else' should be on separate line

File: jdk-1.6-parent/select2-parent/select2/src/main/java/org/wicketstuff/select2/AbstractSelect2Choice.java [hunk 2]
```diff
@@ -257,12 +286,35 @@ public void renderHead(IHeaderResponse response)
...
+			{
+				String raw = getRawInput();
+				return raw == null ? null : convertValue(raw.split(FormComponent.VALUE_SEPARATOR));
+			}
+		} else
```
Comment (solomax) [inline jdk-1.6-parent/select2-parent/select2/src/main/java/org/wicketstuff/select2/AbstractSelect2Choice.java]: same here

File: jdk-1.6-parent/select2-parent/select2/src/main/java/org/wicketstuff/select2/Select2Choice.java [hunk 3]
```diff
@@ -101,4 +96,5 @@ protected void renderInitializationScript(IHeaderResponse response)
 				"$('#%s').select2('data', %s);", getJquerySafeMarkupId(), selection.toJson())));
 		}
 	}
+
```
Comment (solomax) [inline jdk-1.6-parent/select2-parent/select2/src/main/java/org/wicketstuff/select2/Select2Choice.java]: redundant empty line

File: jdk-1.6-parent/select2-parent/select2/src/main/java/org/wicketstuff/select2/Select2MultiChoice.java [hunk 4]
```diff
@@ -35,63 +36,48 @@
  */
 public class Select2MultiChoice<T> extends AbstractSelect2Choice<T, Collection<T>>
 {
+
```
Comment (solomax) [inline jdk-1.6-parent/select2-parent/select2/src/main/java/org/wicketstuff/select2/Select2MultiChoice.java]: redundant empty line

File: jdk-1.6-parent/select2-parent/select2/src/main/java/org/wicketstuff/select2/Select2MultiChoice.java [hunk 5]
```diff
@@ -35,63 +36,48 @@
...
-		final Collection<T> choices;
-		if (Strings.isEmpty(input))
+		String value = getRequest().getRequestParameters().getParameterValue(getInputName()).toString();
+		if (Strings.isEmpty(value))
 		{
-			choices = new ArrayList<T>();
-		}
-		else
+			return EMPTY_STRING_ARRAY;
+		} else
```
Comment (solomax) [inline jdk-1.6-parent/select2-parent/select2/src/main/java/org/wicketstuff/select2/Select2MultiChoice.java]: formatting

File: jdk-1.6-parent/select2-parent/select2/src/main/java/org/wicketstuff/select2/AbstractSelect2Choice.java (hunk 1 above)
Comment (shuraa) [inline jdk-1.6-parent/select2-parent/select2/src/main/java/org/wicketstuff/select2/AbstractSelect2Choice.java]: I've used wicket.xml provided by @martin-g :(

**Comment by shuraa** [comment]
@solomax I'll fix all issues and create another PR w/o formatting commits. Thank you for your time.

**Comment by solomax** [comment]
great, thanks a lot!

**Comment by shuraa** [comment]
Okay, here another PR https://github.com/wicketstuff/core/pull/394 :)

**Pull request closed without merging**